CorsixTH/Lua/world.lua

```diff
@@ -2304,6 +2304,43 @@ function World:resetAnimations()
   end
 end
 
+strict_declare_global "staff_initials_cache"
+staff_initials_cache = {}
+
+local function our_concat(t)
+  -- The standard table.concat function doesn't like our userdata strings :(
+  local result = ""
+  for _, s in ipairs(t) do
+    result = result .. s
+  end
+  return result
+end
+
+function World:updateInitialsCache()
+  -- Generate new set of staff name initials in case the language was changed
+  local parts = tostring(our_concat(_S.humanoid_name_starts)
+      .. our_concat(_S.humanoid_name_ends)):sub(33)
+  local initials = {}
+  for uchar in parts:gmatch("([%z\1-\127\194-\244][\128-\191]*)") do
+    initials[#initials + 1] = uchar
+  end
+  staff_initials_cache.initials = initials
+end
+
+-- Change the staff name first letter to one from the current language
+-- from the seed (generated here or in staff_profile.lua)
+function World:localiseInitial(profile)
+  if not profile.name_seed then
+    profile.name_seed = math.random(1, 1000)
+    profile.name_lang = "false"
```

Comment:
String `false` is a little odd. I guess the idea is that it doesn't match any language name, but wouldn't the empty string (or even nil) work just as well for that?